integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py

```diff
@@ -0,0 +1,116 @@
+just for illustrating basic Kedro features.
+
+Delete this when you start working on your own Kedro project.
+"""
+# pylint: disable=invalid-name
+
+import logging
+import matplotlib.pyplot as plt
+import neptune.new as neptune
+import numpy as np
+import pandas as pd
+from scikitplot.metrics import plot_roc_curve, plot_precision_recall_curve
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import accuracy_score
+from sklearn.neural_network import MLPClassifier
+from typing import Any, Dict
+
+
+def train_rf_model(
+        train_x: pd.DataFrame, train_y: pd.DataFrame, parameters: Dict[str, Any]
+):
+    """Node for training Random Forest model"""
+    max_depth = parameters["rf_max_depth"]
+    n_estimators = parameters["rf_n_estimators"]
+    max_features = parameters["rf_max_features"]
+
+    clf = RandomForestClassifier(max_depth=max_depth,
+                                 n_estimators=n_estimators,
+                                 max_features=max_features)
+    clf.fit(train_x, train_y.idxmax(axis=1))
+
+    return clf
+
+
+def train_mlp_model(
+        train_x: pd.DataFrame, train_y: pd.DataFrame, parameters: Dict[str, Any]
+):
+    """Node for training MLP model"""
+    alpha = parameters["mlp_alpha"]
+    max_iter = parameters["mlp_max_iter"]
+
+    clf = MLPClassifier(alpha=alpha,
+                        max_iter=max_iter)
+    clf.fit(train_x, train_y)
+
+    return clf
+
+
+def get_predictions(rf_model: RandomForestClassifier, mlp_model: MLPClassifier,
+                    test_x: pd.DataFrame) -> Dict[str, Any]:
+    """Node for making predictions given a pre-trained model and a test set."""
+    predictions = {}
+    for name, model in zip(['rf', 'mlp'], [rf_model, mlp_model]):
+        y_pred = model.predict_proba(test_x).tolist()
+        predictions[name] = y_pred
+
+    return predictions
+
+
+def evaluate_models(predictions: dict, test_y: pd.DataFrame,
+                    neptune_run: neptune.run.Handler):
+    """Node for evaluating Random Forest and MLP models and creating ROC and Precision-Recall Curves"""
+
+    for name, y_pred in predictions.items():
+        y_true = test_y.to_numpy().argmax(axis=1)
+        y_pred = np.array(y_pred)
+
+        accuracy = accuracy_score(y_true, y_pred.argmax(axis=1).ravel())
+        neptune_run[f'nodes/evaluate_models/metrics/accuracy_{name}'] = accuracy
+
+        fig, ax = plt.subplots()
+        plot_roc_curve(test_y.idxmax(axis=1), y_pred, ax=ax, title=f'ROC curve {name}')
+        neptune_run['nodes/evaluate_models/plots/plot_roc_curve'].log(fig)
+
+        fig, ax = plt.subplots()
+        plot_precision_recall_curve(test_y.idxmax(axis=1), y_pred, ax=ax, title=f'PR curve {name}')
+        neptune_run['nodes/evaluate_models/plots/plot_precision_recall_curve'].log(fig)
+
+
+def ensemble_models(predictions: dict, test_y: pd.DataFrame,
+                    neptune_run: neptune.run.Handler) -> np.ndarray:
```

Comment:
Not sure, but it probably should be
`neptune_run: neptune.run.Run` -> it's neptune Run object that you use for logging.

Reply:
it is actualy a handler afaik,